File: day3_quick.py
import random

def partition(arr, low ,high):
    pivot=arr[high]
    i=low
    for j in range(i,high):
        if (arr[j]<=pivot):
            arr[j],arr[i]=arr[i],arr[j]
            i+=1
    arr[i],arr[high]=arr[high],arr[i]
    return i

def quicksort(arr, low, high):
    if(low<high):
        p=partition(arr,low,high)
        quicksort(arr,low,p-1)
        quicksort(arr,p+1,high)



def display(arr):
    for i in range(n):
        print(arr[i],end=" ")
    print(end="\n")


if __name__=="__main__":
    arr=[]
    n=int(input("Enter the lenght of an array "))
    for i in range(n):
        arr.append(random.randrange(0,101,2))
    print("The orignal array is ")
    display(arr)
    quicksort(arr,0,n-1)
    print("The array after sorting is ")
    display(arr)

File: day3_quick.cpp
#include<iostream>
#define SIZE 50
using namespace std;

void display(int *arr);
void quicksort(int *arr, int l, int r);
int partition(int *arr, int l, int r);

int main()
{
    int arr[SIZE];
    for(int i=0;i<SIZE;i++)
    {
        arr[i]=rand()%101;
    }
    cout<<"The orignal array "<<endl;
    display(arr);
    quicksort(arr,0,SIZE-1);
    cout<<"The sorted array is "<<endl;
    display(arr);
}
void quicksort(int *arr, int l, int r)
{
    if(l<r)
    {
       int p= partition(arr, l,r);
       quicksort(arr, l, p-1);
       quicksort(arr,p+1, r);
    }
}

int partition(int *arr, int l, int r)
{
    int pivot =arr[r];
    int i=l;
    for(int j=l;j<r;j++)
    {
        if(arr[j]<=pivot)
        {
            swap(arr[j],arr[i]);
            i++;
        }
    }
    swap(arr[i],arr[r]);
    return i;
    
}
void display(int *arr)
{
    for(int i=0;i<SIZE;i++)
    {
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}